pkg/services/auth/auth.go

```diff
@@ -81,9 +117,42 @@ func NewAuthService(fluxClient flux.Flux, k8sClient client.Client, providerName
 	}, nil
 }
 
-// SetupDeployKey creates a git.Git client instrumented with existing or generated deploy keys.
+// GetGitProvider returns the GitProvider associated with the AuthService instance
+func (a *authSvc) GetGitProvider() gitproviders.GitProvider {
+	return a.gitProvider
```

Comment:
I'm not seeing where `gitProvider` gets set. Do we need to check for `nil` here?

Reply:
It's now an argument to `NewAuthService`. I guess that could be called with nil but is that something we need to guard against?

Reply:
Ok I must have missed it. I think the constructor is enough protection.